docs/content/config/advanced/kubernetes.md

```diff
@@ -4,513 +4,754 @@ title: 'Advanced | Kubernetes'
 
-            # other
-            - name: tmp-files
-              mountPath: /tmp
-              readOnly: false
+=== "Sensitive Data"
 
-          ports:
-            - name: transfer
-              containerPort: 25
-              protocol: TCP
-            - name: esmtp-implicit
-              containerPort: 465
-              protocol: TCP
-            - name: esmtp-explicit
-              containerPort: 587
-            - name: imap-implicit
-              containerPort: 993
-              protocol: TCP
+    !!! attention "Sensitive Data"
 
-          envFrom:
-            - configMapRef:
-                name: mailserver.environment
+        For storing OpenDKIM keys, TLS certificates or any sort of sensitive data, you should be using `Secret`s. You can mount secrets like `ConfigMap`s and use them the same way.
 
-      restartPolicy: Always
+    The [TLS docs page][docs-tls] provides guidance when it comes to certificates and transport layer security. Always provide sensitive information via `Secrets`.
 
-      volumes:
-        # configuration files
-        - name: files
-          configMap:
-            name: mailserver.files
+=== "`Deployment`"
 
-        # PVCs
-        - name: data
-          persistentVolumeClaim:
-            claimName: data
+    The `Deployment` config is the most complex component.
 
-        # certificates
-        - name: certificates-rsa
-          secret:
-            secretName: mail-tls-certificate-rsa
-            items:
-              - key: tls.key
-                path: tls.key
-              - key: tls.crt
-                path: tls.crt
+    - It instructs Kubernetes how to run the DMS container and how to apply your `ConfigMap`s, persisted storage, etc.
+    - Additional options can be set to enforce runtime security.
 
-        # other
-        - name: tmp-files
-          emptyDir: {}
-```
+    ```yaml
+    ---
+    apiVersion: apps/v1
+    kind: Deployment
 
-### Certificates - An Example
+    metadata:
+      name: mailserver
 
-In this example, we use [`cert-manager`][cert-manager] to supply RSA certificates. You can also supply RSA certificates as fallback certificates, which DMS supports out of the box with `SSL_ALT_CERT_PATH` and `SSL_ALT_KEY_PATH`, and provide ECDSA as the proper certificates.
+      annotations:
+        ignore-check.kube-linter.io/run-as-non-root: >-
+          'mailserver' needs to run as root
+        ignore-check.kube-linter.io/privileged-ports: >-
+          'mailserver' needs privileged ports
+        ignore-check.kube-linter.io/no-read-only-root-fs: >-
+          There are too many files written to make the root FS read-only
 
-```yaml
----
-apiVersion: cert-manager.io/v1
-kind: Certificate
+    spec:
+      replicas: 1
+      selector:
+        matchLabels:
+          app: mailserver
```

Comment:
The docs no longer refer to the single instance decision, [is this worth addressing](https://github.com/docker-mailserver/docker-mailserver/pull/3928#discussion_r1518574291)?

I'm fine if you don't see the point, just double checking. Can always revise in future if user feedback warrants it.

Reply:
I think this is not worth addressing TBH. Most users will grasp the reasons immediately, and those that need more replicas are (should be) well aware of the implications.